examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -85,12 +85,14 @@ CommissioningParameters PairingCommand::GetCommissioningParameters()
 
 CHIP_ERROR PairingCommand::PairWithQRCode(NodeId remoteId)
 {
-    return CurrentCommissioner().PairDevice(remoteId, mOnboardingPayload);
+    CommissioningParameters commissioningParams = GetCommissioningParameters();
+    return CurrentCommissioner().PairDevice(remoteId, mOnboardingPayload, commissioningParams);
```

Comment:
When we set some commissioning parameters and not others with the PairDevice call, what parameters are we leaving untouched?

If there are parameters we must always overwrite, and others we must always preserve, it feels like these should be in separate structures.

Overall, it seems like merging into the commissioning parameters like this is a good way to inadvertently leak state across commissioning sequences.

Reply:
That is a fair point. I actually sorted a way to do this without the Add command. I will update.